File: comgongdaeng/TwoPointer/BOJ_1806.py
import sys
N, S = map(int, input().split())
sequence = list(map(int, sys.stdin.readline().split()))

left = 0
right = 0
ans = 100000
temp_sum = sequence[0]

while left <= right:
    if temp_sum >= S:
        ans = min(ans, right - left + 1)
        temp_sum -= sequence[left]
        left += 1
    else:
        right += 1
        if right < N:
            temp_sum += sequence[right]
        else:
            break

if ans == 100000:
    print(0)
else:
    print(ans)

File: keemsebin/TwoPointer/BOJ_1806.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

rl.on("line", function (line) {
  input.push(line);
}).on("close", function () {
  let [N, S] = input[0].split(" ").map(Number);
  let arr = input[1].split(" ").map(Number);
  let left = 0;
  let right = 0;
  let sum = 0;
  let minLength = Infinity;
  while (true) {
    if (sum >= S) {
      minLength = Math.min(minLength, right - left);
      sum -= arr[left++];
    } else if (right === N) {
      break;
    } else {
      sum += arr[right++];
    }
  }

  if (minLength === Infinity) {
    console.log(0);
  } else {
    console.log(minLength);
  }
});
